src/modules/job-manager/event.c

```diff
@@ -328,10 +328,16 @@ int event_job_action (struct event *event, struct job *job)
                 && !job->perilog_active
                 && !job->start_pending
                 && !job->free_posted) {
-                if (housekeeping_start (ctx->housekeeping,
-                                        job->R_redacted,
-                                        job->id,
-                                        job->userid) < 0)
+
+                /* Only start housekeeping if this is not an alloc_bypass
+                 * job. This avoids sending a free to scheduler even if
+                 * housekeeping is not configured.
+                 */
```

Comment:
Since this comment follows a comment at the top of the block that ends with
> ... it is safe for the job to release its resources to housekeeping.

maybe it would flow better to say
> Release resources to housekeeping, but only if they were allocated by the scheduler.

Reply:
Oh good suggestion, thanks!

Reply:
Made that change and I'll set MWP. Thanks!